operator/slot_ticker/slotticker.go

```diff
@@ -6,83 +6,90 @@ import (
 	"github.com/attestantio/go-eth2-client/spec/phase0"
 )
 
-// The TTicker interface defines a type which can expose a
-// receive-only channel firing slot events.
-type TTicker interface {
-	C() <-chan phase0.Slot
-	Done()
+//go:generate mockgen -package=mocks -destination=./mocks/slotticker.go -source=./slotticker.go
+
+type SlotTickerProvider func() SlotTicker
+
+type SlotTicker interface {
+	Next() <-chan time.Time
+	Slot() phase0.Slot
+}
+
+type ConfigProvider interface {
+	SlotDurationSec() time.Duration
+	GetGenesisTime() time.Time
 }
 
-// SlotTicker is a special ticker for the beacon chain block.
-// The channel emits over the slot interval, and ensures that
-// the ticks are in line with the genesis time. This means that
-// the duration between the ticks and the genesis time are always a
-// multiple of the slot duration.
-// In addition, the channel returns the new slot number.
-type SlotTicker struct {
-	c    chan phase0.Slot
-	done chan struct{}
+type SlotTickerConfig struct {
+	slotDuration time.Duration
+	genesisTime  time.Time
 }
 
-// C returns the ticker channel. Call Cancel afterwards to ensure
-// that the goroutine exits cleanly.
-func (s *SlotTicker) C() <-chan phase0.Slot {
-	return s.c
+func (cfg SlotTickerConfig) SlotDurationSec() time.Duration {
+	return cfg.slotDuration
 }
 
-// Done should be called to clean up the ticker.
-func (s *SlotTicker) Done() {
-	go func() {
-		s.done <- struct{}{}
-	}()
+func (cfg SlotTickerConfig) GetGenesisTime() time.Time {
+	return cfg.genesisTime
 }
 
-// NewSlotTicker starts and returns a new SlotTicker instance.
-func NewSlotTicker(genesisTime time.Time, secondsPerSlot uint64) *SlotTicker {
-	if genesisTime.IsZero() {
-		panic("zero genesis time")
+type slotTicker struct {
+	timer        *time.Timer
+	slotDuration time.Duration
+	genesisTime  time.Time
+	slot         phase0.Slot
+}
+
+func NewSlotTicker(cfgProvider ConfigProvider) SlotTicker {
```

Comment:
`New` is probably enough here.

Reply:
it's a best practice in go to return concrete types. https://duncanleung.com/go-idiom-accept-interfaces-return-types/